src/java/org/apache/cassandra/cql3/statements/SelectStatement.java

```diff
@@ -786,6 +787,11 @@ private ReadQuery getRangeCommand(QueryOptions options, ColumnFilter columnFilte
         ReadQuery command =
             PartitionRangeReadQuery.create(table, nowInSec, columnFilter, rowFilter, limit, new DataRange(keyBounds, clusteringIndexFilter));
 
+        if (command.isTopK())
+        {
+            checkFalse(aggregationSpec != null, TOPK_AGGREGATION_ERROR);
```

Comment:
So in Apache, the check being only in the `execute` method, there is a bug? I haven't tested, but this one here covers also `executeInternal `and `executeLocally` and in Apache the check is in `execute`, so that made me wonder. :-)

Reply:
This is a very good point. I think I did that change while adding the query offset ([CNDB-9541](https://github.com/datastax/cassandra/commit/c84116c9bf5786fe2574e922dcbf7343dbbe10ae#diff-2ffcbaa5b5edfee0453c64897504f4e3d88150de8977e72d604c4bb688cb3d9b)), but I don't remember why. I think it might be because I found validation missing when calling `CQLTester#execute`, due to the problems addressed by [CNDB-12905](https://github.com/riptano/cndb/issues/12905). 

I have reverted that part and updated `SelectOffsetTest` to use coordinator execution. Skipping validation on replicas and internal queries should give us some (tiny) performance gains. Let's see if something else breaks on CI.

Reply:
> but I don't remember why
I was not involved in CNDB-9541, so I am not sure either. If needed, I can try to look into that rather big patch.
Otherwise, the current change makes sense to me. I hope I do not miss anything from the other patch.

Reply:
I’m 83% sure my past self didn’t realize the issue with `CQLTester#execute` and wrongly thought that all versions of `SELECT` execute should be validated. So it’d be great if you can quickly skim through that patch but I think we are good.

Reply:
Actually, I think we can safely move the rest of the validations in `SelectStatement#getRangeCommand` to `#execute` and save us a bunch of replica-side validations.